### PR TITLE
Extend bin/report

### PR DESCRIPTION
**Pull request by jmorrell**
Extends the `bin/report` output with all of the existing metadata we're capturing by looking at project files and pulling stored metadata from the cache

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -148,10 +148,6 @@ install_bins() {
   meta_set "npm-version-request" "$npm_engine"
   meta_set "yarn-version-request" "$yarn_engine"
 
-  meta_set "node-version-request" "$node_engine"
-  meta_set "npm-version-request" "$npm_engine"
-  meta_set "yarn-version-request" "$yarn_engine"
```
Comment (jmorrell) [inline bin/compile]: These lines were accidental duplicates of the ones above

File: bin/report [hunk 2]
```diff
@@ -7,22 +7,154 @@ set -o errexit    # always exit on error
...
+# pull metadata from the build
+kv_pair_string "build_step" "$(meta_get "build-step")"
+kv_pair_string "failure" "$(meta_get "failure")"
+kv_pair_string "cache_status" "$(meta_get "cache-status")"
+kv_pair "build_time" "$(meta_get "build-time")"
+kv_pair "skipped_prune" "$(meta_get "skipped-prune")"
+kv_pair "cached_bower_components" "$(meta_get "cached-bower-components")"
+kv_pair "custom_cache_dirs" "$(meta_get "node-custom-cache-dirs")"
+
+# pull execution data from the build
```
Comment (jmorrell) [inline bin/report]: All of this timing data would probably be better captured in a trace, but we can iterate towards that

File: test/fixtures/complex-scripts/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "type" : "git",
+    "url" : "http://github.com/example/example.git"
+  },
+  "dependencies": {
+    "hashish": "*"
+  },
+  "engines": {
+    "node": "10.x"
+  },
+  "scripts": {
```
Comment (jmorrell) [inline test/fixtures/complex-scripts/package.json]: Examples of scripts that have caused issues with parsing in the past

File: test/run [hunk 4]
```diff
@@ -1296,8 +1320,8 @@ compile() {
 # that the output of bin/compile will not be captured
 compile_and_report() {
   default_process_types_cleanup
-  cache_dir=$(mktmpdir)
-  env_dir=$(mktmpdir)
+  cache_dir=${2:-$(mktmpdir)}
```
Comment (jmorrell) [inline test/run]: It's important for the test that the cache directory used in `bin/compile` is re-used for `bin/report`, but this was using different directories

Comment (jmorrell) [reply test/run]: It mimics what actually happens during the build, and here we are pulling metadata out of the cache. All of the `meta_get` calls [reference a datastore stored in the cache directory](https://github.com/heroku/heroku-buildpack-nodejs/blob/master/lib/metadata.sh#L10)

Comment (danielleadams) [reply test/run]: Gotcha.. makes sense to me 👍

**Pull request merged**